### PR TITLE
Fetch instead of delete from options hash

### PR DESCRIPTION
**Pull request by ryanmccarthypdx**
This PR seeks to address [Issue 451](https://github.com/procore-oss/blueprinter/issues/451)

Checklist:

* [ ] I have updated the necessary documentation
* [x] I have signed off all my commits as required by [DCO](https://github.com/procore-oss/blueprinter/blob/main/CONTRIBUTING.md)
* [x] My build is green

### CODE REVIEW

File: lib/blueprinter/base.rb [hunk 1]
```diff
@@ -153,10 +153,10 @@ def self.association(method, options = {}, &block)
...
         method: method,
-        name: options.delete(:name) || method,
-        extractor: options.delete(:extractor) || AssociationExtractor.new,
-        blueprint: options.delete(:blueprint),
-        view: options.delete(:view) || :default,
+        name: options.fetch(:name) { method },
+        extractor: options.fetch(:extractor) { AssociationExtractor.new },
+        blueprint: options.fetch(:blueprint),
+        view: options.fetch(:view, :default),
         options: options.merge(block: block)
```
Comment (lessthanjacob) [inline lib/blueprinter/base.rb:160]: To preserve behavior here, we may want to exclude the keys that were previously deleted from the `options` hash (even though `Association` merges `view` and `blueprint` back in):

```suggestion
        options: options.except(
          :name,
          :extractor,
          :blueprint,
          :view 
        ).merge(block: block)
```

File: lib/blueprinter/helpers/base_helpers.rb [hunk 2]
```diff
@@ -15,9 +15,9 @@ module SingletonMethods
...
 
       def prepare_for_render(object, options)
-        view_name = options.delete(:view) || :default
-        root = options.delete(:root)
-        meta = options.delete(:meta)
+        view_name = options.fetch(:view, :default)
+        root = options[:root]
+        meta = options[:meta]
         validate_root_and_meta!(root, meta)
         prepare(object, view_name: view_name, local_options: options, root: root, meta: meta)
```
Comment (lessthanjacob) [inline lib/blueprinter/helpers/base_helpers.rb:22]:
```suggestion
        prepare(
          object, 
          view_name: view_name, 
          local_options: options.except(
            :view, 
            :root,
            :meta
          ), 
          root: root, 
          meta: meta
        )
```

**Review by jhollinger — APPROVED** [review]
👍

**Review by lessthanjacob — APPROVED** [review]
:shipit:

**Pull request merged**